server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/change/TypeStateFactoryTest.scala

```diff
@@ -0,0 +1,81 @@
+package org.apache.james.jmap.change
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+import scala.jdk.CollectionConverters._
+
+class TypeStateFactoryTest {
+  val ALL: Set[TypeName] = Set(EmailTypeName, MailboxTypeName, ThreadTypeName, IdentityTypeName, EmailSubmissionTypeName, EmailDeliveryTypeName)
+  val factory: TypeStateFactory = TypeStateFactory(ALL.asJava)
+
+  @Test
+  def parseEmailTypeNameStringShouldReturnRightEmailTypeName(): Unit = {
+    assertThat(factory.parse("Email")).isEqualTo(Right(EmailTypeName))
+  }
+
+  @Test
+  def parseMailboxTypeNameStringShouldReturnRightMailboxTypeName(): Unit = {
+    assertThat(factory.parse("Mailbox")).isEqualTo(Right(MailboxTypeName))
+  }
+
+  @Test
+  def parseThreadTypeNameStringShouldReturnRightThreadTypeName(): Unit = {
+    assertThat(factory.parse("Thread")).isEqualTo(Right(ThreadTypeName))
+  }
+
+  @Test
+  def parseIdentityTypeNameStringShouldReturnRightIdentityTypeName(): Unit = {
+    assertThat(factory.parse("Identity")).isEqualTo(Right(IdentityTypeName))
+  }
+
+  @Test
+  def parseEmailSubmissionTypeNameStringShouldReturnRightEmailSubmissionTypeName(): Unit = {
+    assertThat(factory.parse("EmailSubmission")).isEqualTo(Right(EmailSubmissionTypeName))
+  }
+
+  @Test
+  def parseEmailDeliveryTypeNameStringShouldReturnRightEmailDeliveryTypeName(): Unit = {
+    assertThat(factory.parse("EmailDelivery")).isEqualTo(Right(EmailDeliveryTypeName))
+  }
+
+  @Test
+  def parseWrongTypeNameStringShouldReturnLeftIllegalArgumentException(): Unit = {
+    assertThat(factory.parse("email").fold(
```

Comment:
You can do asserts `.isInstanceOf(classOf[Left])` then you can drop all the calls to .fold

Also please consider using SoftAssertions....